cogs/general/general.py

```diff
@@ -0,0 +1,433 @@
+        else:
+            await ctx.send("That doesn't look like a question.")
+
+    @commands.command(aliases=["sw"])
+    async def stopwatch(self, ctx):
+        """Starts/stops stopwatch"""
+        author = ctx.message.author
+        if not author.id in self.stopwatches:
+            self.stopwatches[author.id] = int(time.perf_counter())
+            await ctx.send(author.mention + " Stopwatch started!")
+        else:
+            tmp = abs(self.stopwatches[author.id] - int(time.perf_counter()))
+            tmp = str(datetime.timedelta(seconds=tmp))
+            await ctx.send(author.mention + " Stopwatch stopped! Time: **" + tmp + "**")
+            self.stopwatches.pop(author.id, None)
+
+    @commands.command()
+    async def lmgtfy(self, ctx, *, search_terms : str):
+        """Creates a lmgtfy link"""
+        search_terms = escape_mass_mentions(search_terms.replace(" ", "+"))
+        await ctx.send("https://lmgtfy.com/?q={}".format(search_terms))
+
+    @commands.command(hidden=True)
+    @commands.guild_only()
+    async def hug(self, ctx, user : discord.Member, intensity : int=1):
+        """Because everyone likes hugs
+
+        Up to 10 intensity levels."""
+        name = italics(user.display_name)
+        if intensity <= 0:
+            msg = "(っ˘̩╭╮˘̩)っ" + name
+        elif intensity <= 3:
+            msg = "(っ´▽｀)っ" + name
+        elif intensity <= 6:
+            msg = "╰(*´︶`*)╯" + name
+        elif intensity <= 9:
+            msg = "(つ≧▽≦)つ" + name
+        elif intensity >= 10:
+            msg = "(づ￣ ³￣)づ{} ⊂(´・ω・｀⊂)".format(name)
+        await ctx.send(msg)
+
+    @commands.command()
+    @commands.guild_only()
+    async def userinfo(self, ctx, *, user: discord.Member=None):
+        """Shows users's informations"""
+        author = ctx.message.author
+        guild = ctx.message.guild
+
+        if not user:
+            user = author
+
+        roles = [x.name for x in user.roles if x.name != "@everyone"]
+
+        joined_at = self.fetch_joined_at(user, guild)
+        since_created = (ctx.message.timestamp - user.created_at).days
+        since_joined = (ctx.message.timestamp - joined_at).days
+        user_joined = joined_at.strftime("%d %b %Y %H:%M")
+        user_created = user.created_at.strftime("%d %b %Y %H:%M")
+        member_number = sorted(guild.members,
+                               key=lambda m: m.joined_at).index(user) + 1
+
+        created_on = "{}\n({} days ago)".format(user_created, since_created)
+        joined_on = "{}\n({} days ago)".format(user_joined, since_joined)
+
+        game = "Chilling in {} status".format(user.status)
+
+        if user.game is None:
+            pass
+        elif user.game.url is None:
+            game = "Playing {}".format(user.game)
+        else:
+            game = "Streaming: [{}]({})".format(user.game, user.game.url)
+
+        if roles:
+            roles = sorted(roles, key=[x.name for x in guild.role_hierarchy
```

Comment:
The first role in the hierarchy is the everyone role. You can remove it by doing something as basic as `guild.role_hierarchy[1:]`. I also don't know what this is supposed to be doing.

If you want to sort the user's roles by hierarchy the only thing you have to do is `sorted(member.roles)`.